File: demo/Blackholio/client-ts/src/game/FoodController.ts
import Phaser from 'phaser';
import type { Entity, Food } from '../module_bindings/types';
import { EntityController } from './EntityController';

const COLOR_PALETTE = [
  0x77fcad, 0x4cfa92, 0x23f678, 0x77fbc9, 0x4cf9b8, 0x23f5a5,
];

export class FoodController extends EntityController {
  constructor(scene: Phaser.Scene, entity: Entity, food: Food) {
    super(
      scene,
      entity,
      COLOR_PALETTE[Math.abs(food.entityId) % COLOR_PALETTE.length]
    );
  }
}

File: demo/Blackholio/client-godot/FoodController.cs
using Godot;
using SpacetimeDB.Types;

public partial class FoodController : EntityController
{
    private static readonly Color[] ColorPalette =
    [
        new(119 / 255.0f, 252 / 255.0f, 173 / 255.0f),
        new(76 / 255.0f, 250 / 255.0f, 146 / 255.0f),
        new(35 / 255.0f, 246 / 255.0f, 120 / 255.0f),
        new(119 / 255.0f, 251 / 255.0f, 201 / 255.0f),
        new(76 / 255.0f, 249 / 255.0f, 184 / 255.0f),
        new(35 / 255.0f, 245 / 255.0f, 165 / 255.0f),
    ];

    public FoodController(Food food) : base(food.EntityId, ColorPalette[food.EntityId % ColorPalette.Length])
    {
        VisualStyle = CircleVisualStyle.Food;
    }
}
